internal/mode/static/nginx/config/main_config.go

```diff
@@ -32,3 +38,88 @@ func executeMainConfig(conf dataplane.Configuration) []executeResult {
 
 	return results
 }
+
+type mgmtConf struct {
+	Endpoint          string
+	Resolver          string
+	LicenseTokenFile  string
+	DeploymentCtxFile string
+	CACertFile        string
+	ClientSSLCertFile string
+	ClientSSLKeyFile  string
+	SkipVerify        bool
+}
+
+// generateMgmtFiles generates the NGINX Plus configuration file for the mgmt block. As part of this,
+// it writes the secret and deployment context files that are referenced in the mgmt block.
+func (g GeneratorImpl) generateMgmtFiles(conf dataplane.Configuration) []file.File {
+	if !g.plus {
+		return nil
+	}
+
+	tokenFile := file.File{
+		Content: conf.AuxiliarySecrets[graph.PlusReportJWTToken],
```

Comment:
Is this guaranteed to exist? I'm guessing if it doesn't exist, it's due to a programming error. I think it's better to catch this panic and add context to it. Otherwise, the panic will be impossible to trace.